MushroomObserver/mushroom-observer
# Make it so name trackers require approval to send spam notification to observer.

Adds name_trackers.approved column, defaulting to true for all existing name_trackers with non-empty note_template.

When new name_trackers are created with a note_template, it starts as unapproved, and sends a notification to the admins.  The admin notification has a link to approve the name_tracker.

All unapproved name_tracker notifications to the observer are ignored.

A message is displayed for the tracker letting them know that their request is pending approval.

They can check if it has been approved in "Your Interests".

-----

Manual test:

1. go to any random observation
2. go to "About xxx" (name page)
3. click on "Email Tracking" in upper right
4. check the "Send a research request to the observer? " checkbox to enable the spammy half of name tracking, then submit the form

You should see a green flash notice letting you know that your request to send notifications to observers is pending approval, along with a link to "Your Interests" where you can see all of your name_tracker notification requests.  On that page you should see your new name_tracker with the note "(Observer notification pending approval.) "

5. grab the name_tracker id from hovering over the "Change to Ignore" or "Destroy" links (not the "Show Name Tracker" link -- that one has the name id instead of the name_tracker id)
6. type in http://localhost:3000/name/approve_tracker/<id> (inserting the appropriate id from 5)
7. now go back to "Your Interests"

The "pending approval" business shouldn't be there any more.  Note that it does a flash notice "Name stalker approved" when an admin approves of a name_tracker.  If you are logged in as a user who is not an admin it will fail.  But you do not have to actually be in admin mode when you request the /name/approve_tracker action.
